bundles/org.openhab.binding.draytonwiser/src/main/java/org/openhab/binding/draytonwiser/internal/handler/DraytonWiserThingHandler.java

```diff
@@ -0,0 +1,222 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.draytonwiser.internal.handler;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.util.ThingHandlerHelper;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.draytonwiser.internal.DraytonWiserRefreshListener;
+import org.openhab.binding.draytonwiser.internal.api.DraytonWiserApi;
+import org.openhab.binding.draytonwiser.internal.api.DraytonWiserApiException;
+import org.openhab.binding.draytonwiser.internal.model.DraytonWiserDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DraytonWiserThingHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Andrew Schofield - Initial contribution
+ * @author Hilbrand Bouwkamp - Moved generic code from subclasses to this class
+ */
+@NonNullByDefault
+abstract class DraytonWiserThingHandler<T> extends BaseThingHandler implements DraytonWiserRefreshListener {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private @Nullable DraytonWiserApi api;
+    private @Nullable T data;
+    private @Nullable DraytonWiserDTO draytonWiseDTO;
+    private @Nullable ScheduledFuture<?> handleCommandRefreshFuture;
+
+    protected DraytonWiserThingHandler(final Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        final HeatHubHandler bridgeHandler = getHeatHubHandler();
+
+        if (bridgeHandler == null) {
+            api = null;
```

Comment:
Better set the Thing status to OFFLINE with an appropriate error message. Otherwise the Thing will stick with INITIALIZING.

Reply:
The method `getHeatHubHandler()` already sets it to offline in this case.